### DISCUSSION TITLE
which user object is so expensive in performance output?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#25517] [Q&A General]
I've got a mechanics problem that I'm running with PerfGraph output on and heaviest_branch = true. The bottom-most line on the heaviest branch output is "ComputeUserObjects". Is there a way to see which user object is taking up the most time?

**Comment by lindsayad** [comment]
Maybe not with just MOOSE. For this kind of thing I'd use a full-blown profiler like gperftools as described in our profiling [page](https://mooseframework.inl.gov/application_development/profiling.html).

**Reply by jessecarterMOOSE** [reply]
Ah thanks. I'll look into this. Any other profiling tools that you can recommend that may already be installed on an HPC?

**Reply by GiudGiud** [reply]
On INL HPC, gperftools/2.9 is probably the easiest to get going with. Just recompile in oprof mode 

you can also find a vTune module from intel that is quite nice for very advanced studies. You can use it without the intel compilers too

There's a pprof module too
pprof: pprof/2022

**Reply by jessecarterMOOSE** [reply]
I'm on my company's HPC, which is airgapped :disappointed:

**Reply by GiudGiud** [reply]
any of these 3 installed already?

**Reply by jessecarterMOOSE** [reply]
Let me check...

**Reply by jessecarterMOOSE** [reply]
Followup: the `FEProblem::computeUserObjects` method includes Postprocessors too, right?

**Reply by GiudGiud** [reply]
yes
